src/nodes.coffee

```diff
@@ -1750,9 +1768,15 @@ exports.JSXElement = class JSXElement extends Base
       'JSXElement'
 
   elementAstProperties: (o) ->
+    tagNameAst = =>
+      tag = @tagName.unwrap()
+      if tag?.value and ':' in tag.value
+        tag = new JSXNamespacedName tag
```

Comment:
Here the approach is just to continue to pass the namespaced name (containing `:`) as a single token through the grammar and then "parse" it (create new node class instances corresponding to the namespace prefix and name with correct location data) during AST generation

This avoids needing to add any new grammar rules